model/windowInformation.cc

```diff
@@ -67,44 +71,68 @@ namespace minsky
   }
 #endif
 
-  unsigned long WindowInformation::getChildWindowId() {
-    return childWindowId;
-  }
-
-  Display *WindowInformation::getDisplay() {
+  Display *WindowInformation::getDisplay()
+  {
     return display;
   }
 
-  WindowInformation::~WindowInformation() {
-    childSurface.reset();
+  WindowInformation::~WindowInformation()
+  {
+    windowSurface.reset();
+    bufferSurface.reset();
     XDestroyWindow(display, childWindowId);
   }
 
-  ecolab::cairo::SurfacePtr WindowInformation::getSurface() {
-    return childSurface;
+  ecolab::cairo::SurfacePtr WindowInformation::getBufferSurface()
+  {
+    return bufferSurface;
   }
 
-  void WindowInformation::createSurface(){
+  void WindowInformation::copyBufferToMain()
+  {
+    cairo_set_source_surface(windowSurface->cairo(), bufferSurface->surface(), 0, 0);
+    cairo_paint(windowSurface->cairo());
+  }
+
+  void WindowInformation::createSurfaces()
+  {
 #ifdef USE_WIN32_SURFACE
-      {
-          /* TODO */
-      }
+    {
+      /* TODO */
+    }
 #elif defined(MAC_OSX_TK)
 
-      {
-          /* TODO */
-      }
+    {
+      /* TODO */
+    }
 #else
-      {childSurface.reset(new cairo::Surface(cairo_xlib_surface_create(getDisplay(), getChildWindowId(), wAttr.visual, childWidth, childHeight), childWidth, childHeight));
-  cairo_surface_set_device_offset(childSurface->surface(), -wAttr.x, -wAttr.y);
-}
+    {
+      windowSurface.reset(new cairo::Surface(cairo_xlib_surface_create(getDisplay(), childWindowId, wAttr.visual, childWidth, childHeight), childWidth, childHeight));
+      cairo_surface_set_device_offset(windowSurface->surface(), -wAttr.x, -wAttr.y);
+
+
+      bufferSurface.reset(new cairo::Surface(cairo_surface_create_similar(windowSurface->surface(), CAIRO_CONTENT_COLOR_ALPHA, childWidth, childHeight), childWidth, childHeight));
```

Comment:
Does this paint to the same XWindow as the original, or a new child window? The Cairo documentation doesn't give much of a clue.